lib/new_relic/telemetry_sdk/buffer.rb

```diff
@@ -18,6 +18,9 @@ def initialize common_attributes=nil
       # Items recorded into the buffer must have a to_h method for transformation
       def record item
         @lock.synchronize { @items << item }
+      rescue => e
+        logger.error "Encountered error while recording in buffer"
+        logger.error e
```

Comment:
Was this tested?  I don't see unit tests and don't think this would work as written without also including the Logger module in the class.

Reply:
I don't think so either. Good catch.

I also wonder if this means we should revisit our design patterns surrounding the Logger module, like we discussed before. Perhaps it could be more intuitive after all.

Reply:
This is addressed in #32. 😄

Reply:
I had earlier marked this as resolved but I do think we should add the module in to stay in accordance with our existing pattern and then the new pattern can be adopted in #32.

Reply:
Alright, I've added the module and made some tests to make sure each logging method is being checked.